### ISSUE TITLE
NPE on AssertionRemover.removeAssertion

### ISSUE THREAD

**Issue by vmassol** [STAMP-project/dspot#692]
When executing DSpot 1.2.4-SNAPSHOT on commons-filter-api, on AbstractFilterDescriptorManagerTest#namedContainer()

```
[INFO] 965 new tests generated
[WARNING] Too many tests have been generated: 965
[INFO] Number of generated test reduced to 200
[WARNING] Something bad happened during assertion amplification
java.lang.NullPointerException
        at eu.stamp_project.dspot.assertgenerator.AssertionRemover.removeAssertion(AssertionRemover.java:85)
        at eu.stamp_project.dspot.assertgenerator.AssertionRemover.lambda$removeAssertion$0(AssertionRemover.java:47)
        at java.util.stream.ReferencePipeline$7$1.accept(ReferencePipeline.java:267)
        at java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1382)
        at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:481)
        at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:471)
        at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:708)
        at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
        at java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:499)
        at eu.stamp_project.dspot.assertgenerator.AssertionRemover.removeAssertion(AssertionRemover.java:48)
        at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:193)
        at java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1382)
        at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:481)
        at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:471)
        at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:708)
        at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
        at java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:499)
        at eu.stamp_project.dspot.assertgenerator.AssertGenerator.assertionAmplification(AssertGenerator.java:63)
        at eu.stamp_project.dspot.Amplification.assertionsAmplification(Amplification.java:187)
        at eu.stamp_project.dspot.Amplification.amplification(Amplification.java:164)
        at eu.stamp_project.dspot.Amplification.amplification(Amplification.java:115)
        at eu.stamp_project.dspot.DSpot._amplify(DSpot.java:274)
        at eu.stamp_project.dspot.DSpot._amplifyTestClass(DSpot.java:265)
        at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:193)
        at java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1382)
        at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:481)
        at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:471)
        at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:708)
        at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
        at java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:499)
        at eu.stamp_project.dspot.DSpot._amplifyTestClasses(DSpot.java:261)
        at eu.stamp_project.dspot.DSpot.amplifyAllTests(DSpot.java:147)
        at eu.stamp_project.Main.run(Main.java:59)
        at eu.stamp_project.DSpotMojo.execute(DSpotMojo.java:292)
        at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:137)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:210)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:156)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:148)
        at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:117)
        at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:81)
        at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:56)
        at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
        at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:305)
        at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:192)
        at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:105)
        at org.apache.maven.cli.MavenCli.execute(MavenCli.java:956)
        at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:288)
        at org.apache.maven.cli.MavenCli.main(MavenCli.java:192)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
        at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
        at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
        at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)

[INFO] iteration 2 / 3
[INFO] 200 tests selected to be amplified over 200 available tests
[INFO] Amplification of inputs...
 43% |============================================== ]\java.lang.NullPointerException
 47% |============================================== ]-java.lang.NullPointerException
 50% |============================================== ]/java.lang.NullPointerException
 65% |================================================= ]-
````

and hanging...

**Comment by danglotb** [comment]
Could you give me the list of amplifiers you used?

**Comment by vmassol** [comment]
Those ones (without the comments and the ReplacementAmplifier which is not failing ;)):

**Comment by danglotb** [comment]
Fixed with #698 However, your use case still raise some issue (mostly compilation problems in the amplified test methods). 

I'll investigate more on this use case.
